cmd/thanos/receive.go

```diff
@@ -74,17 +72,9 @@ func registerReceive(app *extkingpin.App) {
 			AllowOverlappingBlocks: conf.tsdbAllowOverlappingBlocks,
 		}
 
-		// Local is empty, so try to generate a local endpoint
-		// based on the hostname and the listening port.
-		if conf.endpoint == "" {
-			hostname, err := os.Hostname()
-			if hostname == "" || err != nil {
-				return errors.New("--receive.local-endpoint is empty and host could not be determined.")
-			}
-			parts := strings.Split(*conf.grpcBindAddr, ":")
-			port := parts[len(parts)-1]
-			conf.endpoint = net.JoinHostPort(hostname, port)
-		}
+		// enable ingestion if endpoint is specified, or both the hashrings config are empty,
```

Comment:
Just a nit: let's break up the sentences to make sure that the logic in the comment is crystal clear:
```suggestion
		// Enable ingestion if endpoint is specified or if both the hashrings configs are empty.
		// Otherwise, run the receiver exclusively as a distributor.
```